android-o/AutofillFramework/AutofillFramework/multidatasetservice/AutofillFieldMetadata.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using Android.Service.Autofill;
+using Android.Text;
+using Android.Views;
+using Android.Views.Autofill;
+using static Android.App.Assist.AssistStructure;
+
+namespace AutofillFramework
+{
+	/// <summary>
+	/// A stripped down version of a {@link ViewNode} that contains only autofill-relevant metadata. It
+	/// also contains a {@code mSaveType} flag that is calculated based on the {@link ViewNode}]'s
+	/// autofill hints.
+	/// </summary>
+	public class AutofillFieldMetadata
+	{
+		public SaveDataType SaveType { get; set; }
+
+		public string[] AutofillHints { get; set; }
+
+		public AutofillId AutofillId { get; }
+		public int AutofillType { get; }
```

Comment:
Let's use Android.Views.AutofillType instead of int here to avoid casting later.

Reply:
Yeah, but doing that we should cast the following line: https://github.com/xamarin/monodroid-samples/pull/209/files/9b95497991b57de5bbd100db045c9cc9c9fb4f22#diff-dda2c39a14c5945d8d491a867b352448R29

`view.AutofillType` returns a type of `Android.Text.InputTypes` and it should be converted to `Android.Views.AutofillType`

![screen shot 2017-09-20 at 3 25 41 pm](https://user-images.githubusercontent.com/14874769/30660770-5d7050d8-9e18-11e7-8b42-675ceebd7a50.png)

Reply:
Hmm, this looks like a place where we may want to try to improve enumification, this enum association seems incorrect.

Reply:
Filed this bug, https://bugzilla.xamarin.com/show_bug.cgi?id=59625, which will hopefully be fixed for 15.4 stable.